### PR TITLE
CI

### PR DESCRIPTION
**Pull request by dehall**
CI for testing PRs and auto-deploying to maven central. See https://github.com/mcode/process-message-operation/pull/1 for more details since the contents are the same.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-gradle-jdk8-${{ hashFiles('**/*.gradle*') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-jdk8
+            
+      - name: Create Artifacts
+        run: ./gradlew assemble
+        # just to ensure it compiles; if it doesn't then don't continue
+
+      - name: Configure GPG Key Properties
```
Comment (blangley28) [inline .github/workflows/deploy.yml]: Not that is really matters, but should this be PGP instead of GPG?

**Review by blangley28 — COMMENTED** [review]
As someone who knows nothing about pushing to maven central, is there a way to test this or we'll find out when it's merged? Will it automatically increment the version, how does it know if its a +1, +0.1, or +0.0.1?

**Comment by dehall** [comment]
So unfortunately it's not completely manual. I should have put that in the description too. There are a couple things we still have to do:
1. We decide when to publish a new version and what the new version number is. The number it uses is listed in `build.gradle`: https://github.com/mcode/smart-backend-auth/blob/master/build.gradle#L57
2. The deploy Action runs when we push a new git tag. It doesn't matter what the tag is called. There may be a way to link the version number to the tag, I'm not sure.
3. After the action runs, the artifacts get put into a "staging repository" before they make it to maven central. Manual intervention is required to "release" those and then there is another process that validates a few requirements, and then finally the artifacts get synced to maven central. This is what the staging repo currently looks like, with the most recent deploy action: 
image
(you'll notice the duplicate entries, which are a maven central issue and I don't know if there's anything we can do to prevent that)

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (dehall) [reply .github/workflows/deploy.yml]: Originally it was using GPG (GnuPG) to wrangle the key but that got too complicated. I can remove the term altogether

**Comment by blangley28** [comment]
@dehall sounds good. Would you mind just throwing something in the readme so there's a note to reach out to you

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (blangley28) [reply .github/workflows/deploy.yml]: Ah gotcha, I saw `useInMemoryPgpKeys` and had only heard of PGP keys not GPG keys so I just wanted to make sure

**Review by blangley28 — APPROVED** [review]

**Pull request merged**